### PR TITLE
akobiavi code review 1

### PR DESCRIPTION
**Pull request by JellyV**
**Analysis of the program.**
The purpose of the program is to help you find nearby recycling centers for specific materials, such as batteries, cell phones and so on.

**Was the program available in UC Github on time?**
Yes.

**Is the program documented/commented well enough for you to understand?**
Yes, the code was self-documenting as the variables, classes, and method names were named intuitively so there was no need for extensive documentation.

**Does the program compile?**
Yes, and the tests pass as well.

**Rationale behind your changes.**
There were no logical or syntax errors in the application, and everything they've done so far works just fine. The changes I made mostly focused on best kotlin conventions. For instance, using proper spacing, indentation, using simplified operator (== instead of .equals), and also corrected some typos.

**Links to three specific commits that you made to your own group's GitHub repository before the end of sprint deadline.**
1. Added DTOs and dependencies
 https://github.com/JellyV/FreeWays/commit/9e358de8ad04d73e5b0e9b8a4d6729f1079f7570
2. Created integration tests
https://github.com/JellyV/FreeWays/commit/dad90cff5f3df6853fbc2d16e43a4383dbee5204
3. Created unit test
https://github.com/JellyV/FreeWays/commit/ee542cd4f36aa42555c69616022475e82156630b

**Things I learned**
- I read more about kotlin conventions and learned the best practises. ex: using proper spacing, when to put space before a colon and when not. https://kotlinlang.org/docs/coding-conventions.html#colon
- Adding various elements on the UI
- Implementing Koin dependency injection

### CODE REVIEW

File: app/src/main/java/com/findmyrecycling/ProfileScreen.kt [hunk 1]
```diff
@@ -29,9 +29,9 @@ class ProfileScreen : ComponentActivity() {
 }
 @Composable
 fun ProfileOptions(name: String) {
-    var profile by remember{ mutableStateOf("")}
-    var mySavedLocations by remember{ mutableStateOf("")}
-    var mySavedSearches by remember{ mutableStateOf("")}
+    var profile by remember{ mutableStateOf("") }
+    var mySavedLocations by remember{ mutableStateOf("") }
+    var mySavedSearches by remember{ mutableStateOf("") }
```
Comment (JellyV) [inline app/src/main/java/com/findmyrecycling/ProfileScreen.kt:34]: Added spacing before curly braces (here and other places as well), as per Kotlin convention.
https://kotlinlang.org/docs/coding-conventions.html#lambdas

File: app/src/main/java/com/findmyrecycling/dto/Product.kt [hunk 2]
```diff
@@ -2,7 +2,12 @@ package com.findmyrecycling.dto
 
 import com.google.gson.annotations.SerializedName
 
-data class Product(var product : String,var id : Int = 0, @SerializedName("name")var common : String ) {
+data class Product(
+    var product : String,
+    var id : Int = 0,
+    @SerializedName("name")
+    var common : String,
+    ) {
```
Comment (JellyV) [inline app/src/main/java/com/findmyrecycling/dto/Product.kt:10]: As this class has multiple parameters, I separated it on multiple lines for better readability.
https://kotlinlang.org/docs/coding-conventions.html#class-headers

File: app/src/test/java/com/findmyrecycling/ProductTests.kt [hunk 3]
```diff
@@ -39,7 +39,7 @@ class ProductTests {
     }
 
     @After
-    fun TearDown(){
+    fun tearDown(){
```
Comment (JellyV) [inline app/src/test/java/com/findmyrecycling/ProductTests.kt:42]: function names should start with lowercase

File: app/src/test/java/com/findmyrecycling/ProductTests.kt [hunk 4]
```diff
@@ -50,14 +50,14 @@ class ProductTests {
         thenTheProductCollectionShouldContainCellPhones()
     }
     @Test
-    fun `Given product data are available when I search for batteries then I should receive a location of recycling center having lead acid batteries or rechargable batteries` () = runTest {
+    fun `Given product data are available when I search for batteries then I should receive a location of recycling center having lead acid batteries or rechargeable batteries` () = runTest {
```
Comment (JellyV) [inline app/src/test/java/com/findmyrecycling/ProductTests.kt:53]: fixed a typo

File: app/src/test/java/com/findmyrecycling/ProductTests.kt [hunk 5]
```diff
@@ -76,7 +76,7 @@ class ProductTests {
         assertTrue(allProducts!!.isNotEmpty())
         var containsCellPhones = false
         allProducts!!.forEach {
-            if (it.product.equals("Cell Phones")) {
+            if (it.product == ("Cell Phones")) {
```
Comment (JellyV) [inline app/src/test/java/com/findmyrecycling/ProductTests.kt:79]: (Did this in many places): Replaces .equals with ==, per Android studio's recommendation. It is a simpler syntax
https://kotlinlang.org/docs/equality.html

File: app/src/test/java/com/findmyrecycling/ProductTests.kt [hunk 6]
```diff
@@ -88,7 +88,7 @@ class ProductTests {
         assertTrue(allProducts!!.isNotEmpty())
         var containsBatteries = false
         allProducts!!.forEach {
-            if (it.product.equals("Rechargable Batteries") || it.product.equals("Lead Acid Batteries")) {
+            if (it.product == ("Rechargeable Batteries") || it.product == ("Lead Acid Batteries")) {
```
Comment (JellyV) [inline app/src/test/java/com/findmyrecycling/ProductTests.kt:91]: fixed a typo; replaced .equals with ==

File: app/src/test/java/com/findmyrecycling/ProductTests.kt (hunk 3 above)
Comment (discospiff) [reply app/src/test/java/com/findmyrecycling/ProductTests.kt:42]: Yes, in the case of unit tests, they should.

File: app/src/test/java/com/findmyrecycling/ProductTests.kt (hunk 4 above)
Comment (discospiff) [reply app/src/test/java/com/findmyrecycling/ProductTests.kt:53]: Good catch!

**Comment by discospiff** [comment]
Good suggestions from this review, which can decrease technical debt.  This branch is worth merging.

**Pull request merged**